modules/apps/29-fee/keeper/escrow.go

```diff
@@ -0,0 +1,132 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/cosmos/ibc-go/modules/apps/29-fee/types"
+	channeltypes "github.com/cosmos/ibc-go/modules/core/04-channel/types"
+)
+
+// TODO: add optional relayers arr
+func (k Keeper) EscrowPacketFee(ctx sdk.Context, refundAcc sdk.AccAddress, fee types.Fee, packetID channeltypes.PacketId) error {
+	// check if the refund account exists
+	hasRefundAcc := k.authKeeper.GetAccount(ctx, refundAcc)
+	if hasRefundAcc == nil {
+		return sdkerrors.Wrap(types.ErrRefundAccNotFound, fmt.Sprintf("Account with address: %s not found", refundAcc.String()))
+	}
+
+	fees := sdk.Coins{
+		*fee.AckFee, *fee.ReceiveFee, *fee.TimeoutFee,
+	}
+
+	// check if refundAcc has balance for each fee
+	for _, f := range fees {
+		hasBalance := k.bankKeeper.HasBalance(ctx, refundAcc, f)
+		if !hasBalance {
+			return sdkerrors.Wrap(types.ErrBalanceNotFound, fmt.Sprintf("%s", refundAcc.String()))
+		}
+	}
+
+	for _, coin := range fees {
+		// escrow each fee with account module
+		if err := k.bankKeeper.SendCoinsFromAccountToModule(
```

Comment:
@AdityaSripal I think we should probably do these sends atomically, so only send if all three are successful. Otherwise, we run into the same problem as below with the PayFee/PayFeeTimeout

Reply:
You don't have to worry about that with the SDK. SDK transactions are already atomic (including multimsg txs). If you return an error during tx execution, all state changes made during tx execution get reverted